test/object-store/sync/flx_sync.cpp

```diff
@@ -847,8 +848,16 @@ TEST_CASE("flx: client reset", "[sync][flx][app][baas][client reset]") {
             })
             ->on_post_reset([&, err_future = std::move(error_future)](SharedRealm) mutable {
                 auto sync_error = wait_for_future(std::move(err_future)).get();
-                CHECK(sync_error.get_system_error() ==
-                      sync::make_error_code(sync::ClientError::auto_client_reset_failure));
+                INFO(sync_error.reason());
+                // There is a race here depending on if the server produces a query error or responds to
```

Comment:
@danieltabacaru with respect to https://github.com/realm/realm-core/pull/6711, I am consistently getting a `bad_query` error here when I run the tests locally, even if I use master. Do you have any insight to why that could be, or is it safe to add back this test condition? Was this fixed on the server recently and I have to update my baas version?

Reply:
Yes, it was fixed on the server so you should run a recent baas version.

Reply:
Thanks, using an up to date baas server works for me.